pkg/ring/ring.go

```diff
@@ -32,9 +32,10 @@ type ReadRing interface {
 	prometheus.Collector
 
 	// Get returns n (or more) ingesters which form the replicas for the given key.
-	// buf is a slice to be overwritten for the return value
-	// to avoid memory allocation; can be nil.
-	Get(key uint32, op Operation, buf []IngesterDesc) (ReplicationSet, error)
+	// ingeters and tokens are slices to be overwritten for the return value
+	// to avoid memory allocations; can be nil.
+	Get(key uint32, op Operation, ingesters []IngesterDesc,
+		tokens []TokenDesc) (ReplicationSet, error)
```

Comment:
> Why the line break? it doesn't look very nice.

I've been adding line breaks >80 columns in most cases so it fits on my tiny terminal window without scrolling horizontally. I can merge this into one line if it's that much of an issue. 

> What is 'tokens' parameter used for? What is expected value after calling this method? Documentation missing.

It says it right at the top: 

> // ingeters and tokens are slices to be overwritten for the return value
> // to avoid memory allocations; can be nil.

Reply:
> I've been adding line breaks >80 columns

My personal perspective: it's 2019, and the screen resolution goes well beyond 80 characters and restricting it to 80 is worse than not restricting at all. I personally see this kind of line breaks harder to read.

Reply:
I agree 80 as a hard limit is a bit low, but I do think in general long lines should be broken up. I don't want to try suggesting what a soft limit should be, though. For now, I'm going to change this into one line.

In the case of other functions whose signatures are getting unwieldy, would either of you find one parameter per line easier to read?

```go
Get(
  key uint32,
  op Operation,
  ingesters []IngesterDesc,
  tokens []TokenDesc,
) (ReplicationSet, error) 
```

Reply:
(moved under this thread, my email comment was previously top-level)

First of all, I want to apologize for language used in my previous comment. I wanted to say the same thing as Marco did, but he did it in a much nicer way.

My IDE shows a column line at column 120, and I rarely see code that far (although there are lines crossing it in Cortex). If method signature got that long, I would first reconsider if it has to be like that. If so, I don’t have strong preference between the two styles [although slightly I prefer to use horizontal space, more than vertical], as long as I can see all arguments at onces, without scrolling. I agree with you that horizontal scrolling is annoying.

Reply:
> In the case of other functions whose signatures are getting unwieldy, would either of you find one parameter per line easier to read?

The oneline version of your example is 100 columns wide. I still don't see the real need to split it.

```
Get(key uint32, op Operation, ingesters []IngesterDesc, tokens []TokenDesc) (ReplicationSet, error)
```

From my **personal** perspective: either this max columns is enforced somehow (and tested with the linter), or it's not a thing at all, otherwise we'll end up having different contributors adopting different limits. In the long term, consistency pays off more than perfection. Given the relatively long history of Cortex and the large code base, I would argue this is not a real issue worth to invest our effort, as far as we're OK relaxing the columns limit and we all write code within a reasonable width (ie. the current state of the code base looks good to me).

Reply:
I was using the `Get` function as my example just to demonstrate another way one could shorten the length of a function signature; I was not trying to present it as it being too long, sorry. 

When I split lines at roughly 80 chars, I'm doing it as part of my style; i.e., I find code easier to consume the less I have to move my eyes horizontally (and my terminal window usually isn't very wide either). I don't think we should enforce a particular opinion on the ideal line length just as we shouldn't enforce when one should remove or add a blank line for vertical spacing. It's clearly going to be subjective when a line is "too long," and I agree that this conversation isn't worth our time right now.

My only concern here is doing line breaks in a way that doesn't make the code overall harder for readers to consume. You'll find that there are many places within this PR where I've used line breaks to split up long lines, particularly when calling functions. If you come across these and find them hard to read, please point them out, and I will find another solution to keep the lines short. 

For now, I will improve the documentation for `Get` and merge its function signature into one line.